### PR TITLE
No db lookup when checking for disabled namespaces

### PR DESCRIPTION
**Pull request by gsheasby**
**Goals (and why)**:

==COMMIT_MSG==
Avoid refetching disabled namespaces on every request.
==COMMIT_MSG==

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**: 🔥

### CODE REVIEW

File: timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/DisabledNamespaces.java [hunk 1]
```diff
@@ -57,17 +60,19 @@ public static DisabledNamespaces create(DataSource dataSource) {
 
     private void initialize() {
         execute(Queries::createTable);
+        disabledNamespaces = disabledNamespaces();
```
Comment (jeremyk-91) [inline timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/DisabledNamespaces.java:63]: The javadoc of `Collectors.toSet()` says we don't get guarantees on mutability nor thread-safety, so we can't really do the adds or removes later.

Also, do we expect disable and enable to need to be threadsafe?

Comment (gsheasby) [reply timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/DisabledNamespaces.java:63]: fix for modifiiability:
`                .collect(Collectors.toCollection(HashSet::new));`

Comment (gmaretic) [reply timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/DisabledNamespaces.java:63]: yeah just commented this offline

Comment (gmaretic) [reply timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/DisabledNamespaces.java:63]: `Collectors.toCollection(Sets::newConcurrentHashSet)`

**Pull request merged**